src/test/java/bio/terra/workspace/service/datareference/DataReferenceServiceTest.java

```diff
@@ -130,6 +133,52 @@ public void enumerateFailsUnauthorized() {
         () -> dataReferenceService.enumerateDataReferences(workspaceId, 0, 10, USER_REQUEST));
   }
 
+  @Test
+  public void testUpdateDataReference() {
+    String updatedName = "rename";
+    String updatedName2 = "rename_again";
+    String updatedDescription = "updated description";
+    String updatedDescription2 = "updated description again";
+
+    UUID workspaceId = createDefaultWorkspace();
+    DataReferenceRequest request = defaultReferenceRequest(workspaceId).build();
+
+    DataReference ref = dataReferenceService.createDataReference(request, USER_REQUEST);
+
+    Supplier<DataReference> currentReference =
+        () -> dataReferenceService.getDataReference(workspaceId, ref.referenceId(), USER_REQUEST);
+
+    Consumer<UpdateDataReferenceRequestBody> updateReference =
+        updateBody ->
+            dataReferenceService.updateDataReference(
+                workspaceId, ref.referenceId(), updateBody, USER_REQUEST);
+
+    // Validate the reference exists and is readable.
+    assertThat(currentReference.get(), equalTo(ref));
+
+    // Update name only
+    updateReference.accept(new UpdateDataReferenceRequestBody().name(updatedName));
+    DataReference updatedNameReference1 = currentReference.get();
+    assertThat(updatedNameReference1.name(), equalTo(updatedName));
+    assertThat(updatedNameReference1.referenceDescription(), equalTo(ref.referenceDescription()));
+
+    // Update description only
+    updateReference.accept(
+        new UpdateDataReferenceRequestBody().referenceDescription(updatedDescription));
+    DataReference updatedDescriptionReference = currentReference.get();
+    assertThat(updatedDescriptionReference.name(), equalTo(updatedNameReference1.name()));
+    assertThat(updatedDescriptionReference.referenceDescription(), equalTo(updatedDescription));
+
+    // Update both
+    updateReference.accept(
+        new UpdateDataReferenceRequestBody()
+            .name(updatedName2)
+            .referenceDescription(updatedDescription2));
+    DataReference updatedBothReference = currentReference.get();
+    assertThat(updatedBothReference.name(), equalTo(updatedName2));
+    assertThat(updatedBothReference.referenceDescription(), equalTo(updatedDescription2));
+  }
```

Comment:
see previous comment - add case for both null, consider breaking out individual tests